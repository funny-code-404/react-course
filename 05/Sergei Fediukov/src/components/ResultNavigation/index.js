import React from 'react'
import { Link } from 'react-router-dom'

const ResultNavigation = (props) => {
    return (
        <div className='menu_link' >
            { Boolean(props.data.length) &&
                <h1>
                    {props.title}
                </h1>
            }
            <ul >
                {Boolean(props.data.length) &&
                    props.data.map(x =>
                        <li
                            key={x.id}>
                            <Link
                                to={`${props.path}/${x.id}`}>
                                {x.name}
                            </Link>
                        </li>)
                }
            </ul>
        </div>
    )
}
export default ResultNavigation
