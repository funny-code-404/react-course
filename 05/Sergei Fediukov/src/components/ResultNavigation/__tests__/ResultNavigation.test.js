import { configure, shallow } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'

import ResultNavigation from '..'
configure({ adapter: new Adapter() });

describe('Should render correctly "ResultNavigation"', () => {
    const mockFn = jest.fn();

    const props = {
        data: [
            {
                name: 'test name',
                id: 1
            },
            {
                name: 'test name',
                id: 2
            },
        ],
        title: 'test title',
        path: 'test path'

    }
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<ResultNavigation {...props} />)
    })
    it('to Match Snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })
    it('component is defined', () => {
        expect(wrapper).toHaveLength(1)
    })
    it('component has children div with "menu_link" class', () => {
        expect(wrapper.find('.menu_link')).toHaveLength(1)
    })
})