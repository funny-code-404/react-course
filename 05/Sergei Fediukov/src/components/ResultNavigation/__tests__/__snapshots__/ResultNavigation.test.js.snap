// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly "ResultNavigation" to Match Snapshot 1`] = `
<div
  className="menu_link"
>
  <h1>
    test title
  </h1>
  <ul>
    <li
      key="1"
    >
      <Link
        to="test path/1"
      >
        test name
      </Link>
    </li>
    <li
      key="2"
    >
      <Link
        to="test path/2"
      >
        test name
      </Link>
    </li>
  </ul>
</div>
`;
